crates/ruff/src/noqa.rs

```diff
@@ -20,7 +20,7 @@ use crate::rule_redirects::get_redirect_target;
 
 static NOQA_LINE_REGEX: Lazy<Regex> = Lazy::new(|| {
     Regex::new(
-        r"(?P<spaces>\s*)(?P<noqa>(?i:# noqa)(?::\s?(?P<codes>([A-Z]+[0-9]+(?:[,\s]+)?)+))?)",
+        r"(?P<spaces>\s*)(?P<noqa>(?i:# noqa)(?::\s?(?P<codes>(?:[A-Z]+[0-9]+)(?:[,\s]+[A-Z]+[0-9]+)*))?)(?P<spaces_after_noqa>\s*)",
```

Comment:
`spaces_after_noqa` is used to conserve spaces when there is
```python
print(a)  # noqa: E501, F821 # comment
```
which would become this without matching for `spaces_after_noqa`.
```python
print(a)  # noqa: F821# comment
```
Instead of this
```python
print(a)  # noqa: F821 # comment
```
It is also used to have a better error range.